### ISSUE TITLE
[SourceTree] show files that don’t have a domain

### ISSUE THREAD

**Issue by jasonLaster** [devtools-html/debugger.html#3975]
STR:
1. go to this [url](https://voz37vlg5.codesandbox.io/)
2. observer the actions, components, etc are missing from the tree

#### Firefox

![Screen Shot 2017-09-11 at 8.24.46 PM.png](https://shipusercontent.com/110282e1ac3adb477c81647b6f71d0c7/Screen%20Shot%202017-09-11%20at%208.24.46%20PM.png)

#### Chrome

![Screen Shot 2017-09-11 at 8.24.14 PM.png](https://shipusercontent.com/5a1309588b58865b2a44d17484d275d5/Screen%20Shot%202017-09-11%20at%208.24.14%20PM.png)

#### How do we fix this?

It's actually not too bad, we just need to update the `addToTree` algorithm to use the debuggee URL as the default domain.

```diff
diff --git a/src/utils/sources-tree/addToTree.js b/src/utils/sources-tree/addToTree.js
index b62be5d..e707ea8 100644
--- a/src/utils/sources-tree/addToTree.js
+++ b/src/utils/sources-tree/addToTree.js
@@ -109,6 +109,10 @@ export function addToTree(
 ) {
   const url = getURL(source.get("url"));

+  // if (url is missing a domain) {
+  // prepend the debuggee url
+  // }
+
```

```diff
diff --git a/src/utils/sources-tree/tests/addToTree.spec.js b/src/utils/sources-tree/tests/addToTree.spec.js
index 74df6a5..de8c6cf 100644
--- a/src/utils/sources-tree/tests/addToTree.spec.js
+++ b/src/utils/sources-tree/tests/addToTree.spec.js
@@ -192,5 +192,37 @@ describe("sources-tree", () => {
       sources.forEach(source => addToTree(tree, source));
       expect(formatTree(tree)).toMatchSnapshot();
     });
+
+    it("uses debuggeeurl as the default", () => {
+      // this is a small subset of what we had in the test case
+      const sources = [
+        "components/TodoTextInput.js",
+        "components/Header.js",
+        "reducers/index.js",
+        "components/TodoItem.js",
+        "resource://gre/modules/ExtensionContent.jsm",
+        "https://voz37vlg5.codesandbox.io/static/js/components/TodoItem.js",
+        "index.js"
+      ];
+
+      // the snapshot should look something like this
+      `
+        - http://localhost:4242
+          - components
+            - Header.js
+            - TodoTextInput.js
+          - reducers
+            - (index)
+          - (index)
+          - resource://
+            - gre/modules/ExtensionContent.jsm
+      `;
+
+      const domain = "http://localhost:4242";
+      const sources = createSourcesList(testData);
+      const tree = createNode("root", "", []);
+      sources.forEach(source => addToTree(tree, source, domain));
+      expect(formatTree(tree)).toMatchSnapshot();
+    });
   });
 });
```

**Comment by khiloKenshoo** [comment]
Hi @jasonLaster,
Seems that original path is incorrect, it does have hard coded "/statics/js" in the URL path Object, which should removed in order to fix the issue, below screenshot for clarification : 
![image](https://user-images.githubusercontent.com/8362801/30391978-0ad065d8-98c4-11e7-9d35-0a353af47ce6.png)

Regards,
Khilo Jammal

**Comment by jasonLaster** [comment]
Hmm, i guess my question is are these sources in the source tree? If they're in the source tree in static/js that would be okay.

by the way, i know that that the built files are in the tree. We're looking for the original (perhaps webpacked) sources.

**Comment by jasonLaster** [comment]
ahh, maybe this is not as bad as i thought:

i'm not sure about this no domain thing...

![Screen Shot 2017-09-13 at 1.47.41 PM.png](https://shipusercontent.com/21bdc8d233adcd9d9d24f5bf7766bbaf/Screen%20Shot%202017-09-13%20at%201.47.41%20PM.png)

![Screen Shot 2017-09-13 at 1.47.28 PM.png](https://shipusercontent.com/82fbc3aba70c1fae3a27f68d7195daf8/Screen%20Shot%202017-09-13%20at%201.47.28%20PM.png)

**Comment by jasonLaster** [comment]
okay, so i fixed the dependency and it looks like the original sources are in a (no domain) domain :)

Here is a new GOOD URL to work w/ https://znm3qxj923.codesandbox.io/

![Screen Shot 2017-09-13 at 1.50.44 PM.png](https://shipusercontent.com/3884eaf6f64fd1bfecd91572c081849e/Screen%20Shot%202017-09-13%20at%201.50.44%20PM.png)

**Comment by amitzur** [comment]
From what I see the url has a path of `/static/js/reducers/todos.js`. So how could it be residing next to `static/js` and not inside it?

**Comment by jasonLaster** [comment]
there are two sources - a bundled and original. See the files in (no domain)

**Comment by amitzur** [comment]
I'm not getting the same outcome for some reason - no  `(no domain)` for me:
<img width="715" alt="screen shot 2017-09-13 at 11 23 11 pm" src="https://user-images.githubusercontent.com/394320/30398856-91ce692a-98da-11e7-8242-30bbda8483b3.png">

**Comment by jasonLaster** [comment]
Yeah, i actually am not either... I wonder how they got that :)

I think if you do `appStore.getState().sources.sources.toJS()` you'll see the sources that have a url w/o a domain e.g. `/components/App.js`

**Comment by tromey** [comment]
I think this is fixed by https://github.com/devtools-html/devtools-core/pull/698.  I find it a bit difficult to be sure, but at least more sources do show up.

**Comment by hughugsy** [comment]
So, this is fixed by devtools-html/devtools-core#698 ? :)

**Comment by jasonLaster** [comment]
lets see :)

**Comment by codehag** [comment]
this is fixed by devtools-html/devtools-core#698 -- i left comments on the PR
